rpcs3/util/serialization.hpp

```diff
@@ -108,12 +108,12 @@ namespace utils
 		{
 			if (m_is_writing) return;
 
-			const u32 offset1 = ::offset32(first) + sizeof(first);
+			const u32 offset1 = ::offset32(first) + sizeof(T);
 			const u32 offset2 = ::offset32(second);
 
-			AUDIT(offset2 >= offset1);
+			AUDIT(::offset32(first) <= ::offset32(second));
```

Comment:
Why this?

Reply:
This is because of c++20's [[no_unique_address]].